### ISSUE TITLE
[regression] Can't get trips setting an arrival time

### ISSUE THREAD

**Issue by kalon33** [HSLdevcom/digitransit-ui#491]
Using latest OTP and Digitransit from hsldevcom github, I can't get trips using arrival time as a criteria (no route error), whereas it works properly setting a departure time. It worked previously (a few days ago).

**Comment by Macroz** [comment]
Do you have instructions how to repeat this? Perhaps this has already been fixed.

**Comment by kalon33** [comment]
Using latest master from yesterday evening, that still happens ( http://transit.acolytesanonymes.org ). 

Just try to search an itinerary setting an arrival time instead of a departure time, and you will get this problem. 

This works properly with another interface (like this OTP.js based one, using the same OTP server: http://mobileotp.acolytesanonymes.org ) 

----- Mail original -----

De: "Markku Rontu" <notifications@github.com> 
À: "HSLdevcom/digitransit-ui" <digitransit-ui@noreply.github.com> 
Cc: "Nicolas Derive" <kalon33@ubuntu.com> 
Envoyé: Vendredi 15 Avril 2016 10:36:54 
Objet: Re: [HSLdevcom/digitransit-ui] [regression] Can't get trips setting an arrival time (#491) 



Do you have instructions how to repeat this? Perhaps this has already been fixed.

**Comment by Macroz** [comment]
I just tried against master and it seems to work for me.

![screenshot_2016-04-15_11-50-13](https://cloud.githubusercontent.com/assets/823661/14556349/8068c462-0300-11e6-818e-4fca22302b11.png)

I recall one bug where the sent arriveBy was of wrong type (string instead of boolean). However, this should be fixed by now.

**Comment by kalon33** [comment]
Did you tried also with latest OTP from HSL repo? I only did minor changes in digitransit code on my side (mainly for displaying metro number instead of M, and disabling the system that was preventing proper location display when a polygon is not defined in config), and don't understand why it would work for departure time and not for arrival. 

Do you have a link to the commit that fix arriveBy type? I will look at it on my side. 


----- Mail original -----

De: "Markku Rontu" <notifications@github.com> 
À: "HSLdevcom/digitransit-ui" <digitransit-ui@noreply.github.com> 
Cc: "Nicolas Derive" <kalon33@ubuntu.com> 
Envoyé: Vendredi 15 Avril 2016 10:52:29 
Objet: Re: [HSLdevcom/digitransit-ui] [regression] Can't get trips setting an arrival time (#491) 



I just tried against master and it seems to work for me. 




I recall one bug where the sent arriveBy was of wrong type (string instead of boolean). However, this should be fixed by now.

**Comment by Macroz** [comment]
I'm talking about commit 36bb6c04101c286ae8be56e822ff696ee65b0e9a.

Running local OTP HSL master and UI master it works for me.

**Comment by kalon33** [comment]
I've no idea why, but that commit didn't get merged when I did merge master locally on my server. I made a diff last week to track changes, and that problem happened just after... Damn Murphy's law! 

That works properly now. Thanks for your help! 


----- Mail original -----

De: "Markku Rontu" <notifications@github.com> 
À: "HSLdevcom/digitransit-ui" <digitransit-ui@noreply.github.com> 
Cc: "Nicolas Derive" <kalon33@ubuntu.com> 
Envoyé: Vendredi 15 Avril 2016 11:05:34 
Objet: Re: [HSLdevcom/digitransit-ui] [regression] Can't get trips setting an arrival time (#491) 



I'm talking about commit 36bb6c0 . 

Running local OTP HSL master and UI master it works for me.
